File: srv/src/main/java/customer/ai2code/service/TaskService.java
package customer.ai2code.service;

import cds.gen.mainservice.CreateTaskWithBotsContext;
import cds.gen.mainservice.TasksGetHierarchyContext;
import customer.ai2code.model.task.Task;

public interface TaskService {

    public Task createTaskWithBots(CreateTaskWithBotsContext context);

    public Task createTaskWithBots(String name, String description, String taskTypeId);

    public Task createTaskWithBots(String botInstanceId, String name, String description, String contextPath,
            int sequence);

    public Task createTaskWithBots(String botInstanceId, String name);

    /**
     * Get Current Task(with bots) by taskId.
     * 
     * @param taskId
     * @return
     */
    public Task getCurrentTask(String taskId);

    /**
     * Get Current Task(no bots) by botInstanceId(as parent) and sequence.
     * 
     * @param botInstanceId
     * @param sequence
     * @return
     */
    public Task getCurrentTask(String botInstanceId, int sequence);

    public String getHierarchy(String taskId);

    public String getHierarchy(TasksGetHierarchyContext context);

    public String getMainTaskId(String botInstanceId);

    public boolean deleteOriginalTasks(String botInstanceId);
}

File: srv/services/TaskService.ts
import { Task } from "srv/model/Task";
/**
 * 任务服务接口
 */
export interface TaskService {
  createTaskWithBots(context: any): Task;

  createTaskWithBots(
    name: string,
    description: string,
    taskTypeId: string
  ): Task;

  createTaskWithBots(
    botInstanceId: string,
    name: string,
    description: string,
    contextPath: string,
    sequence: number
  ): Task;

  createTaskWithBots(botInstanceId: string, name: string): Task;

  getCurrentTask(taskId: string): Task;

  getCurrentTask(botInstanceId: string, sequence: number): Task;
}
